### ISSUE TITLE
In the Mapping CRD, the timeout_ms parameter is being ignored if set to 0

### ISSUE THREAD

**Issue by suever** [datawire/ambassador#3212]
**Describe the bug**
The documentation for [Ambassador](https://www.getambassador.io/docs/latest/topics/using/timeouts/#request-timeout-timeout_ms) (and envoy) indicates that setting the `timeout_ms` parameter to `0` for a particular route/mapping disables the timeout. When setting this value to `0`, the default timeout (3000 ms) is actually used by ambassador and passed to Envoy. 

I believe this is caused by the `Mapping` CRD being parsed by golang where [the `TimeoutMs` field is tagged with `omitempty` ](https://github.com/datawire/ambassador/blob/master/pkg/api/getambassador.io/v2/mapping_types.go#L78)and the datatype is `int` where a 0-value is considered "empty" by golang and is therefore omitted when serializing the data. ([a quick example](https://play.golang.org/p/JSKXMsbdKmY)).

This can be confirmed by looking at the YAML in the developer console for a `Mapping` where the `timeout_ms` is explicitly set to `0` in the CRD. In the YAML output, you can see that the `timeout_ms` is omitted in the console YAML in this case.

**To Reproduce**
1. Create a `Mapping` with the `timeout_ms` set to zero:

```
apiVersion: getambassador.io/v2
kind: Mapping
metadata:
  name: example
spec:
  prefix: /
  service: service.namespace:80
  timeout_ms: 0
```

2. Create a dummy service that takes longer than 3 seconds (the default timeout) to respond
3. Make a request and see a 504 Gateway Timeout at 3 seconds (the default timeout)
4. Look at the mapping in the Ambassador Console and inspect the output YAML and you will see the following, which explains why the default timeout is being used:

```
apiVersion: getambassador.io/v2
kind: Mapping
metadata:
  name: example
spec:
  prefix: /
  service: service.namespace:80
```

**Expected behavior**
I would expect the `timeout_ms` being set to 0 on the `Mapping` CRD to be respected and no timeout being used by Ambassador/envoy.

**Versions (please complete the following information):**
 - Ambassador: v1.11.1
 - Kubernetes environment: GKE
 - Version 1.18

**Comment by pjediny** [comment]
I observe same behaviour described here. The go `TimeoutMs` field should probably be defined as `*int` to differentiate a default empty value and disabling the timeout. Other timeouts probably have the same problem.

**Comment by kflynn** [comment]
Amazing timing, I've been working on this very issue. 😐  There are several fields with exactly this issue, yes -- I'm trying to get them all at once. I'll post a dev build later today if any of y'all are interested in trying it -- thanks in advance!

**Comment by suever** [comment]
> Amazing timing, I've been working on this very issue. 😐 There are several fields with exactly this issue, yes -- I'm trying to get them all at once. I'll post a dev build later today if any of y'all are interested in trying it -- thanks in advance!

Always happy to try out a fix and provide feedback, @kflynn. Thanks!

**Comment by imranismail** [comment]
I'm facing the same issue, it's not respected

**Comment by kflynn** [comment]
This was fixed in Ambassador 1.11.2, and should've been closed then!
